### DISCUSSION TITLE
Printer with meta :color

### DISCUSSION THREAD

**Discussion by Chemaclass** [phel-lang/phel-lang#168] [Q&A] [answered]
What do you think about being able to define the color of the text by the Printer?
How hard would it be to add metadata to the `print-str` in order to specify some color? 
We could provide a list of default colors like green, yellow, red, white, and black.

I've been thinking/trying something like this
```clojure
(defn print-str
  "Same as print. But instead of writing it to an output stream,
  The resulting string is returned."
  @{:color color}
  [& xs]
  (let [len (count xs)
        printer (php/:: Printer (nonReadable color))
        pp |(php/-> printer (print $))]
    (case (count xs) 
    ... etc ...
```
But the output I get is `Can not resolve symbol 'color'`. I think I am using the metadata wrong 🤔

**Comment by jenshaase** [comment]
The meta data you mentioned is only available for `def` but not for `defn`.

Regarding color: Do you want colorful html output or colorful bash output?

**Reply by Chemaclass** [reply]
I was thinking more about "colorful bash output"

**Comment by Chemaclass** [comment, marked as answer]
This was solved by creating a new function `println-colorful` https://github.com/phel-lang/phel-lang/pull/202
